### ISSUE TITLE
The size is too big, could you shrink down to smaller than 100MB?

### ISSUE THREAD

**Issue by trumhemcut** [jumanjihouse/docker-dotnet#7]

**Comment by jumanjiman** [comment]
i've opened https://github.com/jumanjihouse/docker-dotnet/pull/8 to eliminate every unnecessary dependency. the size gets down to ~~310~~ 300 MiB, but it does not seem possible to get the image down to 100 MiB.

any ideas?

**Comment by trumhemcut** [comment]
How about not using mono, just pure coreclr, it will shrink down much. I'm trying with this script but still issues there.

```
FROM iron/base

RUN echo '@edge http://nl.alpinelinux.org/alpine/edge/main' >> /etc/apk/repositories
RUN echo '@community http://nl.alpinelinux.org/alpine/edge/community' >> /etc/apk/repositories
RUN echo '@testing http://nl.alpinelinux.org/alpine/edge/testing' >> /etc/apk/repositories
RUN apk update && apk upgrade

RUN apk add --update bash

ENV DNX_VERSION 1.0.0-rc1-update1
ENV DNX_USER_HOME /opt/DNX_BRANCH
ENV DNX_RUNTIME_ID ironbase

# Is it needed?
# RUN echo > ~/.bash_profile
# RUN echo > ~/.profile

RUN apk add unzip curl gettext  

# These package is not curretnly available on alpine
# libssl-dev libcurl3-gnutls   sqlite3 libsqlite3-dev libunwind8 libicu-dev zlib1g

RUN curl -sSL https://raw.githubusercontent.com/aspnet/Home/dev/dnvminstall.sh | DNX_USER_HOME=$DNX_USER_HOME DNX_BRANCH=v$DNX_VERSION sh
RUN bash -c "source $DNX_USER_HOME/dnvm/dnvm.sh \
    && dnvm install $DNX_VERSION -alias default -r coreclr \
    && dnvm alias default | xargs -i ln -s $DNX_USER_HOME/runtimes/{} $DNX_USER_HOME/runtimes/default"

ENV PATH $PATH:$DNX_USER_HOME/runtimes/default/bin

# Clean APK cache
RUN rm -rf /var/cache/apk/*
```

**Comment by jumanjiman** [comment]
> How about not using mono, just pure coreclr

is that possible?

https://github.com/dotnet/coreclr/blob/master/Documentation/building/linux-instructions.md#install-mono implies that mono is a hard requirement.

**Comment by jumanjiman** [comment]
i also found http://stackoverflow.com/a/32229101 which states:

> Mono is equivalent to .NET Framework. .NET Core is completely a different thing,
> 
> http://blogs.msdn.com/b/dotnet/archive/2014/12/04/introducing-net-core.aspx
> 
> Your application must be compiled and designed against .NET Core 5 profile so as to be able to run on .NET Core 5. That's not an automatic thing. Currently you can only use Visual Studio 2015 to do such development (or in a code editor with command line tools).
> 
> corerun is described slightly in this sample,
> 
> https://github.com/dotnet/coreclr/blob/master/Documentation/building/osx-instructions.md

**Comment by trumhemcut** [comment]
yeah, that makes sense to me. I'm running .net corelcr app on my Mac without installing mono. That's why I think mono is not necessary in the image. But still struggling with script, I don't have exp. with linux based, just get it familiar with for now, so you may fix it.

**Comment by jumanjiman** [comment]
> yeah, that makes sense to me. I'm running .net corelcr app on my Mac without installing mono. That's why I think mono is not necessary in the image. But still struggling with script, I don't have exp. with linux based, just get it familiar with for now, so you may fix it.

i'll have to play with it and see if i can work your idea into the image.
i have linux, not dotnet experience :grin:

**Comment by jumanjiman** [comment]
> ```
> # These package is not curretnly available on alpine
> # libssl-dev libcurl3-gnutls   sqlite3 libsqlite3-dev libunwind8 libicu-dev zlib1g
> ```

the alpine package database can be useful to identify the alpine versions of these dependencies.
one can use https://pkgs.alpinelinux.org/packages to search for package names, and
one can use https://pkgs.alpinelinux.org/contents to search for package contents.

for example:
- libssl-dev is provided by `openssl-dev` https://pkgs.alpinelinux.org/packages?name=openssl-dev&repo=all&arch=x86_64&maintainer=all
- libicu-dev is provided by `icu-dev` per https://pkgs.alpinelinux.org/contents?filename=%25libicu%25&path=&pkgname=&repo=all&arch=x86_64

**Comment by trumhemcut** [comment]
Thanks @jumanjiman Will trill this as well.

There is one issue if you can help:
- Although I have included `bash -c "source $DNX_USER_HOME/dnvm/dnvm.sh` in the Dockerfile to enable dnvm command, but it doesn't saved the status, I have to run that command again after starting the container to enable dnvm command. Do you have any ideas?

Thanks,
Phi

**Comment by trumhemcut** [comment]
The only idea is that I'm trying to build a new image like this https://github.com/aspnet/aspnet-docker/blob/master/1.0.0-rc1-update1-coreclr/Dockerfile 
but rather than inheriting from debian:jessie, I'd like to use the iron/base (which is a tiny one).
